src/itehandler/IteToSwitch.cc

```diff
@@ -0,0 +1,283 @@
+            } else {
+                for (auto &nodeAndPolarity : myParents) {
+                    PTRef cond = na[nodeAndPolarity.first].getCond();
+                    lbool pol = nodeAndPolarity.second;
+                    PTRef parentCondition = dagNodeToPTRef.at(nodeAndPolarity.first);
+                    PTRef edgeCondition = pol == l_True ? cond : logic.mkNot(cond);
+                    parentConditions.push(logic.mkAnd(parentCondition, edgeCondition));
+                }
+            }
+
+            PTRef pathCondition = logic.mkOr(parentConditions);
+            dagNodeToPTRef[el_r] = pathCondition;
+
+            if (el.isLeaf()) {
+                // A value node
+                switches.push_m(CondValPair(pathCondition, el.getVal()));
+            }
+        }
+    }
+
+    return switches;
+}
+
+ite::Dag IteToSwitch::constructIteDag(PTRef root, const Logic &logic) {
+
+    ite::Dag dag;
+
+    vec<PTRef> queue;
+    vec<type> flag;
+    flag.growTo(logic.getNumberOfTerms());
+
+    queue.push(root);
+    while (queue.size() != 0) {
+        PTRef tr = queue.last();
+        const Pterm &t = logic.getPterm(tr);
+        auto index = Idx(t.getId());
+        if (flag[index] == ite::type::black) {
+            queue.pop();
+            continue;
+        }
+
+        flag[index] = type::gray;
+
+        bool unprocessed_children = false;
+
+        for (int i = 0; i < t.size(); i++) {
+            auto childIndex = Idx(logic.getPterm(t[i]).getId());
+            if (flag[childIndex] == type::white) {
+                queue.push(t[i]);
+                unprocessed_children = true;
+            }
+        }
+
+        if (unprocessed_children) {
+            continue;
+        }
+
+        flag[index] = type::black;
+
+        queue.pop();
+
+        if (logic.isIte(tr)) {
+
+            dag.addItePTRef(tr);
+
+            const Pterm &ite = logic.getPterm(tr);
+            PTRef cond_tr = ite[0];
+            PTRef true_tr = ite[1];
+            PTRef false_tr = ite[2];
+            {
+                ite::NodeRef true_node = dag.getNodeOrCreateLeafNode(true_tr);
+                ite::NodeRef false_node = dag.getNodeOrCreateLeafNode(false_tr);
+                assert(logic.hasSortBool(cond_tr));
+                dag.createAndStoreNode(tr, cond_tr, true_node, false_node);
+            }
+
+        } else { // not Ite
+            for (int i = 0; i < t.size(); i++) {
+                if (logic.isIte(t[i]) and !dag.isTopLevelIte(t[i])) {
+                    // Term t[i] is an ite which appears as a child of a non-ite.
+                    // We store this term for an expansion into a switch.
+                    dag.addTopLevelIte(t[i]);
+                }
+            }
+        }
+    }
+    return dag;
+}
+
+void IteToSwitch::constructSwitches() {
+    vec<PTRef> ites = iteDag.getTopLevelItes();
+
+    for (auto tl_ite : ites) {
+        ite::Dag subDag = iteDag.getReachableSubGraph(logic, tl_ite);
```

Comment:
This line is redundant. The local variable subDag is not used and the same creation of subgraph is repeated in the makeSwitch method.